cpp/include/cudf/detail/iterator.cuh

```diff
@@ -80,6 +81,84 @@ struct null_replaced_value_accessor
   }
 };
 
+/** -------------------------------------------------------------------------*
+ * @brief pair accessor of column with null bitmask
+ * A unary functor returns pair with scalar value at `id` and boolean validity
+ * `operator() (cudf::size_type id)` computes `element` and valid flag at `id`
+ * returns a `pair(element, valid)`
+ * This functor is only allowed for nullable columns.
+ *
+ * the return value for element `i` will return `pair(column[i], true)`  
+ * if it is valid, or `pair(null_replacement, false)` if it is null.
+ *
+ * @throws `cudf::logic_error` if the column is not nullable.
+ * @throws `cudf::logic_error` if column datatype and Element type mismatch.
+ *
+ * @tparam Element The type of elements in the column
+ * -------------------------------------------------------------------------**/
+template <typename Element>
+struct null_replaced_pair_accessor: public null_replaced_value_accessor<Element>
+{
+  /** -------------------------------------------------------------------------*
+   * @brief constructor
+   * @param[in] _col column device view of cudf column
+   * @param[in] null_replacement The value to return for null elements
+   * -------------------------------------------------------------------------**/
+  null_replaced_pair_accessor(column_device_view const& _col, Element null_val)
+      : null_replaced_value_accessor<Element>{_col, null_val}
+  { }
+
+  CUDA_DEVICE_CALLABLE
+  thrust::pair<Element, bool> operator()(cudf::size_type i) const {
+    return this->col.is_valid_nocheck(i)
+               ? thrust::make_pair(this->col.template element<Element>(i), true)
+               : thrust::make_pair(this->null_replacement, false);
+  }
+};
+
+/** -------------------------------------------------------------------------*
+ * @brief pair accessor of column with/without null bitmask
+ * A unary functor returns pair with scalar value at `id` and boolean validity
+ * `operator() (cudf::size_type id)` computes `element`  and
+ * returns a `pair(element, validity)`
+ *
+ * the return value for element `i` will return `pair(column[i], validity)`
+ * `validity` is `true` if `has_nulls=false`.
+ * `validity` is validity of the element at `i` if `has_nulls=true` and the
+ * column is nullable.
+ *
+ * @throws `cudf::logic_error` if `has_nulls==true` and the column is not
+ * nullable.
+ * @throws `cudf::logic_error` if column datatype and Element type mismatch.
+ *
+ * @tparam Element The type of elements in the column
+ * @tparam has_nulls boolean indicating to treat the column is nullable
+ * -------------------------------------------------------------------------**/
+template <typename Element, bool has_nulls=false> 
+struct pair_accessor
+{
+  column_device_view const col;      ///< column view of column in device
```

Comment:
Class data members should be private (Scott Meyers Effective C++ 3rd Ed. Item 22 Declare members private)

Reply:
This code is from a different PR that this one depends on : https://github.com/rapidsai/cudf/pull/3490/files

Reply:
Then please feel free to resolve the conversation. Merged-in code from other PR's should not be the responsibility of the current PR contributor(s). 

(BTW, is there a quick way to see that code that shows as "new changes" in a PR is actually the result of a merge from another PR?)

Reply:
When they are from a different author, yes.  See the image.  Click on "files changed", then use the commits dropdown to select only commits from the current PR author. Then you can review those.

![image](https://user-images.githubusercontent.com/783069/71046722-429e5700-218d-11ea-9d54-50e59c07c118.png)